components/button/button.tsx

```diff
@@ -82,11 +88,23 @@ const Button = React.forwardRef<HTMLButtonElement, React.PropsWithChildren<Butto
       ...props
     } = filteredProps
 
-    const { bg, border, color } = useMemo(() => getButtonColors(theme.palette, filteredProps), [
+    const defaultColors = useMemo(() => getButtonColors(theme.palette, filteredProps), [
       theme.palette,
       filteredProps,
     ])
-    const hover = useMemo(() => getButtonHoverColors(theme.palette, filteredProps), [
+    let hoverColors = useMemo(() => getButtonHoverColors(theme.palette, filteredProps), [
+      theme.palette,
+      filteredProps,
+    ])
+    let activeColors = useMemo(() => getButtonActiveColors(theme.palette, filteredProps), [
+      theme.palette,
+      filteredProps,
+    ])
+    let loadingColors = useMemo(() => getButtonLoadingColors(theme.palette, filteredProps), [
+      theme.palette,
+      filteredProps,
+    ])
+    let disabledColors = useMemo(() => getButtonDisabledColors(theme.palette, filteredProps), [
```

Comment:
`useMemo` may be overused in this repository.

> You may rely on `useMemo` as a performance optimization, not as a semantic guarantee.

Reply:
the four functions can merge into one

Reply:
and if the value does not change, better to use const

Reply:
Done